Mathlib/Algebra/Lie/Submodule.lean

```diff
@@ -66,6 +66,9 @@ instance : AddSubgroupClass (LieSubmodule R L M) M where
   zero_mem N := N.zero_mem'
   neg_mem {N} x hx := show -x ∈ N.toSubmodule from neg_mem hx
 
+instance instSmulMemClass : SMulMemClass (LieSubmodule R L M) R M where
+  smul_mem {s} c _ h := SubMulAction.smul_mem' s.toSubMulAction c h
```

Comment:
hopefully

```suggestion
  smul_mem {s} c _ h := s.smul_mem'  c h
```

or if not

```suggestion
  smul_mem {s} c _ h := s.toSubMulAction.smul_mem' c h
```

but why do we need to do this if we already have a SubMulAction? shouldn't this be getting picked up automatically?

Reply:
We need it so that the lemma `SetLike.mk_smul_mk` applies to terms including terms of type `LieSubmodule R L M`.

It's true that we have projections `LieSubmodule.toSubmodule` and `Submodule.toSubMulAction` but we still need a `SMulMemClass` instance indexed on `LieSubmodule R L M` because though these projections exist, it is the type that is used to look up the typeclass instances.

Note that we have the corresponding instance for `Submodule` [here](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Submodule/Basic.html#Submodule.smulMemClass).

I do however agree that we _shouldn't_ have to write all the boilerplate nonsense.

Reply:
Does `deriving` deal with this? If not maybe we should write a handler for it.

Reply:
Yes, I believe this is a job for `deriving` and I think the handler is missing.